opentelemetry/proto/trace/v1/trace.proto

```diff
@@ -249,6 +249,26 @@ message Span {
   // An optional final status for this span. Semantically when Status isn't set, it means
   // span's status code is unset, i.e. assume STATUS_CODE_UNSET (code = 0).
   Status status = 15;
+
+  // ParentSpanIsRemote indicates whether the span's parent is remote (true / false) or
+  // if the field is unset.
+  enum ParentSpanIsRemote {
+    // Unset. This value is expected from older clients. In which case, ParentSpanIsRemote
+    // can be derived by looking at the parent span and determining if it has a different Resource.
+    PARENT_SPAN_IS_REMOTE_UNSET = 0;
+
+    // The parent span is not remote.
+    PARENT_SPAN_IS_REMOTE_FALSE = 1;
+
+    // The parent span is remote.
+    PARENT_SPAN_IS_REMOTE_TRUE = 2;
+  }
+
+  // parent_span_is_remote identifies whether a span's parent is remote or not.
+  // This helps determine whether a span can be considered an entry-point span.
+  // A span is an entry-point span if it has no parent (parent_span_id is empty),
+  // or if parent_is_remote is true.
+  ParentSpanIsRemote parent_span_is_remote = 16;
```

Comment:
[This comment](https://github.com/open-telemetry/opentelemetry-proto/blob/81a296f9dba23e32d77f46d58c8ea4244a2157a6/opentelemetry/proto/trace/v1/trace.proto#L79) on span indicates the next available field is 17, which suggests 16 has previously been allocated. Should investigate this and use 17 / update the comment if necessary.

Reply:
[It looks like](https://github.com/open-telemetry/opentelemetry-proto/commit/a9a9d064d9592e8a1454ed0d8026c0da7427468f) field 16 was previously used for `local_child_span_count`.

Reply:
We discussed in the 8/22/23 spec SIG and concluded that 16 is the correct field number. Several field numbers including 16 and 17 where used for other purposes prior to the 0.7.0 version. However, 0.7.0 introduced several breaking changes that make it not possible to produce with any version prior to 0.7.0. The thinking is that we should effectively ignore that versions prior to 0.7.0 existed for the purposes of field numbers.

Relevant links:
- [remove local_child_span_count](https://github.com/open-telemetry/opentelemetry-proto/pull/107/files) - merged on  2020/02/19
- [v0.7.0 release](https://github.com/open-telemetry/opentelemetry-proto/releases/tag/v0.7.0) - released on 2021/01/27

This means that the [comment](https://github.com/open-telemetry/opentelemetry-proto/blob/81a296f9dba23e32d77f46d58c8ea4244a2157a6/opentelemetry/proto/trace/v1/trace.proto#L79) indicating that 17 is the next available field is incorrect. The next available field is in fact 16.